class/practice2.2/main.cpp

```diff
@@ -0,0 +1,125 @@
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <sstream>
+#include <algorithm>
+
+using namespace std;
+
+struct Pixel {
+    int r, g, b;
+};
+
+bool isValidPixel(const Pixel &pixel) {
+    return (pixel.r >= 0 && pixel.r <= 255) && (pixel.g >= 0 && pixel.g <= 255) && (pixel.b >= 0 && pixel.b <= 255);
+}
+
+bool loadPixels(const string &filename, vector<vector<Pixel>> &pixels, string &errorMessage) {
+    ifstream file(filename);
+    if (!file.is_open()) {
+        errorMessage = "Error: Could not open file " + filename;
+        return false;
+    }
+
+    string line;
+    int lineCount = 0;
+    while (getline(file, line)) {
+        if (line.empty()) continue; 
+        stringstream ss(line);
+        vector<Pixel> row;
+        string pixelData;
+        while (getline(ss, pixelData, ' ')) {
+            Pixel pixel;
+            replace(pixelData.begin(), pixelData.end(), ',', ' ');
+            stringstream pixelStream(pixelData);
+            if (!(pixelStream >> pixel.r >> pixel.g >> pixel.b) || !isValidPixel(pixel)) {
+                errorMessage = "Invalid pixel data: " + pixelData;
+                return false;
+            }
+            row.push_back(pixel);
+        }
+        if (row.size() != 16) {
```

Comment:
it is better to avoid hardcoded constants(aka magic numbers) across the code. Youu can use constexpr constant in anonymous namespace - it will behave the same but maintenance of such a code will be easier.